src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -75,6 +77,35 @@ public static string CreateWorkItemTitle(this Run run)
                     (locationName == null ? "" : " (in " + locationName + ")");
         }
 
+        public static Dictionary<string, int> FetchToolResultCounts(this SarifLog log)
+        {
+            if (log == null) { throw new NullReferenceException(); }
+            if (log.Runs == null) { throw new ArgumentNullException(nameof(log.Runs)); }
+
+            Dictionary<string, int> resultCountsByTool = new Dictionary<string, int>();
```

Comment:
> Dictionary<string, int> [](start = 11, length = 24)

NIT, our automated code style will convert this explicit type declaration to 'var'.